error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/TimeRules.java

```diff
@@ -142,6 +142,24 @@ OffsetDateTime after(Instant instant, ZoneId zoneId) {
     }
   }
 
+  /** Don't unnecessarily transform an {@link Instant} to an equivalent instance. */
+  static final class InstantIdentity {
+    @BeforeTemplate
+    Instant before(Instant instant) {
+      return Refaster.anyOf(
+          instant.plus(Duration.ZERO),
+          instant.minus(Duration.ZERO),
+          Instant.parse(instant.toString()),
+          Instant.ofEpochMilli(instant.toEpochMilli()),
+          Instant.ofEpochSecond(instant.getEpochSecond()));
```

Comment:
These two aren't actually identity transformations, as they drop the nano and millisecond components, respectively. Let's introduce rules to suggest using `Instant#truncatedTo` instead.

Reply:
Makes sense, thanks 💪